### DISCUSSION TITLE
Windows line endings in toThrowError()

### DISCUSSION THREAD

**Discussion by drwpow** [vitest-dev/vitest#6364] [Q&A]
Need help in the least-smelly way to deal with Windows errors. My tests are full of failures like this:

```
packages/cli test: . test:js: -   4 |       "100": {
packages/cli test: . test:js: +   4 |       "100": {\r
packages/cli test: . test:js: -   5 |         "$type": "color",
packages/cli test: . test:js: +   5 |         "$type": "color",\r
packages/cli test: . test:js: - > 6 |         "$value": { "colorSpace": "srgb", "channels": "[0, 0.2, 1]", "alpha": 1 }
packages/cli test: . test:js: + > 6 |         "$value": { "colorSpace": "srgb", "channels": "[0, 0.2, 1]", "alpha": 1 }\r
packages/cli test: . test:js: -     |                                                       ^
packages/cli test: . test:js: +     |                                                       ^\r
packages/cli test: . test:js: -   7 |       }
packages/cli test: . test:js: +   7 |       }\r
packages/cli test: . test:js: -   8 |     }
```

My code is as simple as:

```ts
await expect(async () => {
  await someAsyncAction();
}).rejects.toThrowError(expectedErrorMsgString)
```

I’ve tried

1. Sidestepping `rejects.toThrowError()` entirely, but then all my tests are in manual `try/catch` blocks with lots of string replacement and I’m not using Vitest conveniences
2. Conditionally changing the `expectedErrorMsgString` line endings based on platform, but that in some ways feels even worse

Also as a general rule I try to **avoid** having branching logic paths in tests (`if (os.platform() === 'win32') { …` because that also can hide bugs.

What are others’ strategies for testing error messages on Windows and Linux reliably?

**Comment by sheremet-va** [comment]
Have you tried forcing a specific eol with gitattributes? Like this:

https://github.com/vitest-dev/vitest/blob/main/.gitattributes

**Reply by drwpow-figma** [reply]
I have not! I’ll give that a try, thank you!
